File: 1586-longest-subarray-of-1s-after-deleting-one-element/1586-longest-subarray-of-1s-after-deleting-one-element.c
int longestSubarray(int* nums, int numsSize){
    int i = 0;
    int j = 0;
    int k = 1;
    int maxlen = 0;

    while (j < numsSize){
        if (nums[j] == 0) k--;
        if (k < 0){
            if (nums[i] == 0) k++;
            i++;
        }
        j++;
        maxlen = fmax(maxlen, j-i-1);
    }
    return maxlen;

}

File: 1586-longest-subarray-of-1s-after-deleting-one-element/1586-longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        i = j = 0
        skip = 1
        maxlen = 0
        while j < len(nums):
            if nums[j] == 0:
                skip -= 1
            if skip < 0:
                if nums[i] == 0:
                    skip += 1
                i += 1
            j+=1
            maxlen = max(maxlen, j-i-1)
        
        return maxlen
